### PR TITLE
Refs #25217 - fix AutoComplete eslint

### PR DESCRIPTION
**Pull request by sharvit**

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteClearButton.js [hunk 1]
```diff
@@ -5,31 +5,26 @@ import { Icon, OverlayTrigger, Tooltip } from 'patternfly-react';
...
-        name="close"
-        className="autocomplete-clear-button"
-        onClick={onClear}
-      />
-    </OverlayTrigger>
-  );
-};
+const AutoCompleteClearButton = ({ onClear }) => (
+  <OverlayTrigger
+    overlay={<Tooltip id={UUID()}>{__('Clear')}</Tooltip>}
```
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteClearButton.js:10]: It is better to generate a new `uuid` in every for each render then use it as a `default-prop`.
when uses as `default-prop` it produce the same id for all the instances.

**Comment by sharvit** [comment]
@Laviro can you review please?

File: webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteClearButton.js (hunk 1 above)
Comment (sharvit) [reply webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteClearButton.js:10]: an alternative is to set it in a class constructor

**Review by Ron-Lavi — APPROVED** [review]
Thanks @sharvit !!
Tested, looks great :+1:

**Review by ohadlevy — APPROVED** [review]

**Pull request merged**